### ISSUE TITLE
Question : I use shadowsocks with cloak as a plugin  , aes-256-gcm ecryption , is it secure enough ?

### ISSUE THREAD

**Issue by abhinavisgood** [lxhao61/integrated-examples#32]
I am living inisde a college university campus , where shadowsocks was blocked by default , what should i do , currently i am using shadowsocks behind Cloak proxy , and its working , but i am afraid that the VPS server's ip is gonna be exposed and gonna be blocked , could You suggest a way to circumvent this situation , and 
(Note: i am currently a student , hence not in a situtation in buying any sort of CDN subscription from amazon or azure for domain fronting )

should i be worried of ip being detected (given that ip itself resolve to google.com only)

**Comment by lxhao61** [comment]
是否安全是对比的，从来没有绝对的安全。我之前也不了解你的这种应用（shadowsocks with cloak as a plugin），从我刚了解 cloak 插件的介绍，觉得应该还无[v2ray(SS+v2ray-plugin)+caddy\nginx](https://github.com/lxhao61/integrated-examples/tree/main/v2ray(SS%2Bv2ray-plugin)%2Bcaddy%5Cnginx) 应用安全及可靠等。

**Comment by abhinavisgood** [comment]
With all due respect , thank you for your answer and i hereby agree with your reply fully , just need an agreement that "no firewall is more powerfull than the GFW , right?" i'm asking this because of the level of security my firewall has i am (may be?? too much)  overwhelmed so that's why i am asking this question , i have worked many nights in reasearch of a perfect firewall circumvention strategy , i am very afraird on letting that reaserch fail, and about cloak , ==>  see this https://github.com/cbeuw/Cloak please ,  please do so see this project i read its documentation , its a really good project .

**Comment by abhinavisgood** [comment]
One more request , i am having problem in implementing , this [caddy(N+T) ] , can you refer me to some documentation or tutorial on how so many caddyfile are to be run , are'nt a single caddy instance only run a single file ,  thats my question , and one more thing that ,
is caddy-trojan plugin a  full fledged trojan proxy in itself , and won't need a seperate trojan process to run  ? 

I am having difficulty in implenting , if you could please , pleeeeeeeeeeeeeeeeease , help me with the implementation  ,   big request ...

Thank you

**Comment by lxhao61** [comment]
> With all due respect , thank you for your answer and i hereby agree with your reply fully , just need an agreement that "no firewall is more powerfull than the GFW , right?" i'm asking this because of the level of security my firewall has i am (may be?? too much) overwhelmed so that's why i am asking this question , i have worked many nights in reasearch of a perfect firewall circumvention strategy , i am very afraird on letting that reaserch fail, and about cloak , ==> see this https://github.com/cbeuw/Cloak please , please do so see this project i read its documentation , its a really good project .

我初步了解了cloak实现原理，没有觉得多好。当然这个cloak可以延续太老的shadowsocks等应用增强安全等，但就代理通过GFW来说，我不推荐使用这个插件来实现。我推荐使用Xray或v2ray、trojan或trojan-go等来实现。

**Comment by lxhao61** [comment]
> One more request , i am having problem in implementing , this [caddy(N+T) ] , can you refer me to some documentation or tutorial on how so many caddyfile are to be run , are'nt a single caddy instance only run a single file , thats my question , and one more thing that , is caddy-trojan plugin a full fledged trojan proxy in itself , and won't need a seperate trojan process to run ?
> 
> I am having difficulty in implenting , if you could please , pleeeeeeeeeeeeeeeeease , help me with the implementation , big request ...
> 
> Thank you

caddy-trojan 插件不能脱离caddy独立使用，即必须提前把插件编译集成到caddy中来使用。你可以直接下载我已经编译集成好的caddy文件来使用。

**Comment by abhinavisgood** [comment]
So That means i have to combine multiple Caddy files into one , right no  ?  , i know it sounds lame , but i have never worked like  , though so its not understood by me   , just want to reconfirm it from you

**Comment by lxhao61** [comment]
> So That means i have to combine multiple Caddy files into one , right no ? , i know it sounds lame , but i have never worked like , though so its not understood by me , just want to reconfirm it from you

是的。你下载编译好的集成caddy程序文件即可。https://github.com/lxhao61/integrated-examples/releases

**Comment by abhinavisgood** [comment]
I'm having difficulty assembling the caddy file , i don't know which cadyfile has to be combined 

this is my best possible work  , and its showing error  , please tell why , and i am trying to setup a redirection url , whenever someone tries to type in myip it rejects it and whenever someone types my domain he is redirected to someting like




{
	order trojan before map
	order forward_proxy before trojan
	admin off
	log {
		output file /var/log/caddy/access.log
		level ERROR
	}
	servers :443 {
		listener_wrappers {
			trojan
		}
		protocol {
			allow_h2c
			experimental_http3
		}
	}
}

:443, girly.gq, endpoint1.transport.girly.gq, endpoint2.transport.girly.gq {
	tls {
		ciphers TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
		alpn h2 http/1.1
	}

	forward_proxy {
		basic_auth admin d0UX3gwJR0niwWOAZDKAyn
		hide_ip
		hide_via
		probe_resistance
	}

	trojan {
		user 6jyhCN4HiiN9ahUZ3EYfNA
		connect_method
		websocket #开启Websocket支持
	}

	@host {
		host girly.gq endpoint1.transport.girly.gq endpoint2.transport.girly.gq
	}
	route @host {
		header {
			Strict-Transport-Security "max-age=31536000; includeSubDomains; preload" #启用HSTS
		}
		file_server {
			root /var/www/html
		}
	}
}
{
@host {
	host girly.gq #Limited domain name access (prohibit access to the website by ip), change to your own domain name. 
}
route @host {
	header {
		Strict-Transport-Security "max-age=31536000; includeSubDomains; preload" #启用HSTS 
	}
	file_server {
		root /var/www/html #Change to the path of the web file stored by yourself 
	}
}

@host1 {
	host google.transport.girly.gq
}
route @host1 {
	header {
		Strict-Transport-Security "max-age=31536000; includeSubDomains; preload"
	}
	file_server {
		root /var/www/za
	}
	redir https://www.youtube.com/
}
}
